cpp/src/io/parquet/reader_impl_chunking.cu

```diff
@@ -753,82 +755,97 @@ std::vector<row_range> compute_page_splits_by_row(device_span<cumulative_page_in
 }
 
 /**
- * @brief Decompresses a set of pages contained in the set of chunks.
+ * @brief Stores basic information about pages compressed with a specific codec.
+ */
+struct codec_stats {
+  Compression compression_type  = UNCOMPRESSED;
+  size_t num_pages              = 0;
+  int32_t max_decompressed_size = 0;
+  size_t total_decomp_size      = 0;
+
+  enum class page_selection { DICT_PAGES, NON_DICT_PAGES };
+
+  void add_pages(host_span<ColumnChunkDesc const> chunks,
+                 host_span<PageInfo> pages,
+                 page_selection selection)
+  {
+    for (auto& page : pages) {
+      if (chunks[page.chunk_idx].codec == compression_type &&
+          (page.flags & PAGEINFO_FLAGS_DICTIONARY) == (selection == page_selection::DICT_PAGES)) {
```

Comment:
Nonblocking nit: can we make this logic a bit easy to read using a variable? Maybe, `auto is_dict_pages_selected = selection == page_selection::DICT_PAGES`